File: Common/Example03_console_input/java/src/Input03.java
// read multiple console inputs and print the sum to the console
import java.util.Scanner;

public class Input03 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter numbers (space-separated) Type Ctrl + D to end:  ");
        int sum = 0;
        while (scanner.hasNextInt()) {
            int num = scanner.nextInt();
            sum += num;
        }
        System.out.println("Sum: " + sum);
        scanner.close();
    }
}

File: Common/Example03_console_input/c/src/Input03.c
// read multiple console inputs and print the sum to the console
#include<stdio.h>

int main(){
    int sum = 0, num;
    printf("Enter numbers (space-separated): ");
    while(scanf("%d", &num) == 1){ // scanf returns the number of successfully read items
        sum = sum + num;

        if (getchar() == '\n') { // check newline for end of input
            break;
        }
    }
    printf("Sum: %d\n", sum);
    return 0;
}